### DISCUSSION TITLE
Issue when I am calling getUserLocale function in i18n.js file to get the locale from cookies

### DISCUSSION THREAD

**Discussion by vishal-rana1310** [amannn/next-intl#1194] [Q&A]
I am getting this error
Error: Route / with `dynamic = "error"` couldn't be rendered statically because it used `cookies`.
Basically getRequestConfig method doesn't have access to coookies

what should i do?

i18n.js file

`import { getRequestConfig } from "next-intl/server";
import { getUserLocale } from "./services/locale";
export default getRequestConfig(async () => {
  // Provide a static locale, fetch a user setting,
  // read from `cookies()`, `headers()`, etc.

  const locale = await getUserLocale();
  // const locale = "en"
  return {
    locale,
    messages: (await import(`../messages/${locale}.json`)).default,
  };
});`

locale.js file

`"use server";
import { getCookies } from "@/i18n";
import { cookies } from "next/headers";
const COOKIE_NAME = "NEXT_LOCALE";
import { defaultLocale } from "../config";
export async function getUserLocale() {
  const locale = cookies().get(COOKIE_NAME).value; // determine locale from cookies or other logic
  return locale;
}
export async function setUserLocale(locale) {
  await getCookies();
  cookies().set(COOKIE_NAME, locale);
}
`

**Comment by amannn** [comment]
If you're calling `cookies`, then Next.js requires dynamic rendering—that's just how it works.

Note that there's an option to setup [with i18n routing](https://next-intl-docs.vercel.app/docs/getting-started/app-router) and then use [`localePrefix. 'never'`](https://next-intl-docs.vercel.app/docs/routing#locale-prefix-never). That's an option to have cookie-based locales while keeping static rendering if that's very important to you. I generally recommend the setup without i18n routing though if you don't need the `[locale]` segment to be visible to the user.
